src/LifxNet/Payload.cs

```diff
@@ -0,0 +1,422 @@
+		/// </summary>
+		/// <returns>byte</returns>
+		public byte GetUint8() {
+			try {
+				return _br.ReadByte();
+			} catch {
+				Debug.WriteLine("Error reading byte, pos is " + _ms.Position);
+			}
+
+			return 0;
+		}
+
+		/// <summary>
+		/// Read UInt16 from array and increment pointer 2 bytes
+		/// </summary>
+		/// <returns>ushort</returns>
+		public ushort GetUInt16() {
+			try {
+				return _br.ReadUInt16();
+			} catch {
+				Debug.WriteLine($"Error getting Uint16 from payload, pointer {_ms.Position} of range: " + _len);
+			}
+
+			return 0;
+		}
+
+		/// <summary>
+		/// Read Int16 from array and increment pointer 2 bytes.
+		/// </summary>
+		/// <returns>short</returns>
+		public short GetInt16() {
+			try {
+				return _br.ReadInt16();
+			} catch {
+				Debug.WriteLine($"Error getting int16 from payload, pointer {_ms.Position} of range: " + _len);
+			}
+
+			return 0;
+		}
+
+		/// <summary>
+		/// Read Int32 from array and increment pointer 4 bytes.
+		/// </summary>
+		/// <returns>int</returns>
+		public int GetInt32() {
+			try {
+				return _br.ReadInt32();
+			} catch {
+				Debug.WriteLine($"Error getting Int32 from payload, pointer {_ms.Position} of range: " + _len);
+			}
+
+			return 0;
+		}
+
+		/// <summary>
+		/// Read a UInt32 from array and increment pointer 4 bytes.
+		/// </summary>
+		/// <returns></returns>
+		public uint GetUInt32() {
+			try {
+				return _br.ReadUInt32();
+			} catch {
+				Debug.WriteLine($"Error getting Uint32 from payload, pointer {_ms.Position} of range: " + _len);
+			}
+
+			return 0;
+		}
+
+		/// <summary>
+		/// Read an Int64 from array and increment pointer 8 bytes.
+		/// </summary>
+		/// <returns>long</returns>
+		public long GetInt64() {
+			try {
+				return _br.ReadInt64();
+			} catch {
+				Debug.WriteLine($"Error getting Int64 from payload, pointer {_ms.Position} of range: " + _len);
+			}
+
+			return 0;
+		}
+
+		/// <summary>
+		/// Read a UInt64 from array and increment pointer 8 bytes.
+		/// </summary>
+		/// <returns>ulong</returns>
+		public ulong GetUInt64() {
+			try {
+				return _br.ReadUInt64();
+			} catch {
+				Debug.WriteLine($"Error getting Uint64 from payload, pointer {_ms.Position} of range: " + _len);
+			}
+
+			return 0;
+		}
+
+		/// <summary>
+		/// Read a Float32 from array and increment pointer 4 bytes.
+		/// </summary>
+		/// <returns>float</returns>
+		public float GetFloat32() {
+			try {
+				return _br.ReadSingle();
+			} catch {
+				Debug.WriteLine($"Error getting Float32 from payload, pointer {_ms.Position} of range: " + _len);
+			}
+
+			return 0;
+		}
+
+		/// <summary>
+		/// Read a string from our payload.
+		/// </summary>
+		/// <param name="length">The number of chars to read. If none specified, will read the entire payload</param>
+		/// <returns>string</returns>
+		public string GetString(long length = -1) {
+			if (length == -1) length = _len - 1 - _ms.Position;
+			try {
+				return _br.ReadChars((int) length).ToString();
```

Comment:
I think that this  should be `return new string(_br.ReadChars((int) length));`  as it stands, this line returns `System.Char[]{}` when it's called via `LifxClient.GetDeviceLabelAsync()`